### ISSUE TITLE
‘list index out of range’ error

### ISSUE THREAD

**Issue by BKLi** [marianogabitto/ChromA#6]
Hello,

When running ChromA we ran into an error:

```python
 File "/path/to/models.py", line 222, in save_bedfile
    if self.annotations[0].shape[1] > 2:
IndexError: list index out of range
```
Could you maybe let us know what might be causing this problem?

Thank you!

**Comment by marianogabitto** [comment]
Hello,
     would it be possible that you paste the content of the .log files and the command you are using to run ChromA?

Thanks,

Mariano

**Comment by BKLi** [comment]
Hi Mariano, 

I've attached the log file and the command I used was 

```bash
ChromA -i LM059.sorted.nodup.bam -sb LM059.sorted.nodup.downsample.chroma.bed -spec human
```
The sample was in hg38.

btw I noticed in the doc that the version should be 0.1.0 but somehow I got version 0.0.5 and my coworker got 0.0.9 -- do you think if that might be relevant?

Thanks!

[LM059.sorted.nodup.downsample.chroma.bed.log](https://github.com/marianogabitto/ChromA/files/4840201/LM059.sorted.nodup.downsample.chroma.bed.log)

**Comment by marianogabitto** [comment]
Dear Bink,
       I found strange that some chromosomes do not find information to annotate. Would it be possible to ask you if there are some empty chromosomes in your sample?

M

**Comment by BKLi** [comment]
Hmmm... I don't think so -- we did sequence the whole genome. Okay maybe let me try it on some other data and I'll get back to you when I finish.

**Comment by marianogabitto** [comment]
Do you have an email to correspond to?

M

**Comment by BKLi** [comment]
Yup -- libingkun@stu.scu.edu.cn

**Comment by BKLi** [comment]
So I tested two other datasets, both of which are whole-genome data, and got the same errors. I attached the log for both test runs.

Thanks!
[test2.sorted.chroma.bed.log](https://github.com/marianogabitto/ChromA/files/4840466/test2.sorted.chroma.bed.log)
[test3.sorted.chroma.bed.log](https://github.com/marianogabitto/ChromA/files/4840467/test3.sorted.chroma.bed.log)

**Comment by marianogabitto** [comment]
This error was due to processing ATAC-seq experiments with single-ended reads. ChromA was not originally developed to process single-end reads, it was developed to process paired-end reads. The commonly observed insert size distribution plots can only be computed from paired-end reads.

We have adapted ChromA to recognize single-ended reads and keep email communication with Bingkun.

Thanks!

M
